core/fetcher.go

```diff
@@ -0,0 +1,83 @@
+package core
+
+import (
+	"context"
+	"fmt"
+
+	rpctypes "github.com/celestiaorg/celestia-core/rpc/core/types"
+	"github.com/celestiaorg/celestia-core/types"
+	"github.com/celestiaorg/celestia-node/service/block"
+)
+
+const newBlockSubscriber = "NewBlock/Events"
+
+var newBlockEventQuery = types.QueryForEvent(types.EventNewBlock).String()
+
+type BlockFetcher struct {
+	Client
+
+	newBlockCh chan *block.Raw
+}
+
+// NewBlockFetcher returns a new `BlockFetcher`.
+func NewBlockFetcher(client Client) *BlockFetcher {
+	return &BlockFetcher{
+		Client: client,
+	}
+}
+
+// GetBlock queries Core for a `Block` at the given height.
+func (f *BlockFetcher) GetBlock(ctx context.Context, height *int64) (*block.Raw, error) {
+	raw, err := f.Block(ctx, height)
+	return raw.Block, err
+}
+
+// SubscribeNewBlockEvent subscribes to new block events from Core, returning
+// a new block event channel on success.
+func (f *BlockFetcher) SubscribeNewBlockEvent(ctx context.Context) (<-chan *block.Raw, error) {
+	// start the client if not started yet
+	if !f.IsRunning() {
+		if err := f.Start(); err != nil {
+			return nil, err
+		}
+	}
```

Comment:
I see the intention here, but: 
* we defined semantics for Start/Stop and if some code didn't do Start that means a bug, while this code will hide it from us.
* BlockFetcher is only responsible for fetching blocks, not for managing the lifecycle of the client
* We should not overcomplicate BlockFecther with unnecessary things.

Reply:
Instead, we should just check if the Client is running and error back if not.